processor/otel/consumer.go

```diff
@@ -81,14 +82,43 @@ func (c *Consumer) convert(td consumerdata.TraceData) *model.Batch {
 			continue
 		}
 
-		var parentID *string
 		root := len(otelSpan.ParentSpanId) == 0
-		if !root {
-			str := fmt.Sprintf("%x", otelSpan.ParentSpanId)
-			parentID = &str
+
+		var parentID, spanID, traceID string
+		if td.SourceFormat == sourceFormatJaeger {
+			if !root {
+				jaegerParentSpanID, err := tracetranslator.BytesToUInt64SpanID(otelSpan.ParentSpanId)
```

Comment:
Could you please extract this and the conversion of `otelSpan.SpanId` below into a separate function? Something like:

```go
func formatJaegerSpanID(id []byte) string {...}
```

And in that function, please add a comment with a link to https://github.com/jaegertracing/jaeger-client-go/blob/v2.23.1/span_context.go - or more canonical documentation, if you're aware of any.